frame/dapps-staking/src/pallet/mod.rs

```diff
@@ -588,9 +604,29 @@ pub mod pallet {
         /// However, caller have to have deposit amount.
         /// TODO: weight
         #[pallet::weight(T::WeightInfo::payout_stakers_alive_staked(T::MaxStakings::get()))]
-        pub fn register(origin: OriginFor<T>, contract_id: T::AccountId) -> DispatchResult {
-            // TODO: impls
-            Ok(())
+        pub fn register(
+            origin: OriginFor<T>,
+            contract_id: SmartContract<T::AccountId>,
+        ) -> DispatchResultWithPostInfo {
+            let operator = ensure_signed(origin)?;
+            let mut ledger = Self::ledger(&operator).ok_or(Error::<T>::NotController)?;
```

Comment:
The method documentation says 'Any user can call this function'.

Does it mean that stash account should also be able to call this?

Reply:
Let's first see if we are going to support 2 accounts for the  contract registration and if there will be 2 payout calls. One for claiming staker's reward and one for claiming developer's reward.

Reply:
Ok, sure.

It'd be good to have these 'open questions/issues' in the IP document you're preparing.

Also for stuff like 'We only support 1 contract per person' information and similar stuff.